From a519c4d3777347fcc2fb53e20595d230eac41708 Mon Sep 17 00:00:00 2001
From: Michel Thebeau <Michel.Thebeau@windriver.com>
Date: Tue, 26 Aug 2025 20:18:02 +0000
Subject: [PATCH] set password policy regex, unique count

A pre-starlingx change to set the platform's password regex and
unique_last_password_count. Was squashed with misc other puppet-keystone
changes. Splitting the change for clarity. Original git log:

Adds support for Keystone password rules. Password rules were introduced
in Openstack Newton as a [compliance] requirement. We enable the
following rules, in line with other password rules enforced [on the
platform]:
- Password must be a minimum of 7 characters
- Password Complexity: 1 lower + 1 upper + 1 digit + 1 special
- Password History: 2 in chamber

Rebased onto puppet-keystone 25.0.0-3
Signed-off-by: Michel Thebeau <Michel.Thebeau@windriver.com>

Original sign-offs of the squashed patch; rebases and refactors:
Signed-off-by: Dan Voiculeasa <dan.voiculeasa@windriver.com>
Signed-off-by: Dean Troyer <dtroyer@gmail.com>
Signed-off-by: Al Bailey <Al.Bailey@windriver.com>

Original sign-off:
Signed-off-by:  Kam Nasim <kam.nasim@windriver.com>
---
 spec/classes/keystone_security_compliance_spec.rb | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/spec/classes/keystone_security_compliance_spec.rb b/spec/classes/keystone_security_compliance_spec.rb
index 4856f3f..4287476 100644
--- a/spec/classes/keystone_security_compliance_spec.rb
+++ b/spec/classes/keystone_security_compliance_spec.rb
@@ -23,9 +23,9 @@ describe 'keystone::security_compliance' do
           :lockout_failure_attempts           => 3,
           :minimum_password_age               => 4,
           :password_expires_days              => 5,
-          :password_regex                     => 'SomeRegex',
-          :password_regex_description         => 'this is some regex',
-          :unique_last_password_count         => 6,
+          :password_regex                     => '^(?=.*\d)(?=.*[a-zA-Z]).{7,}$',
+          :password_regex_description         => 'password must be at least 7 characters long and contain 1 digit',
+          :unique_last_password_count         => 2,
         }
       end
       it 'should have configure security compliance with params' do
@@ -35,9 +35,9 @@ describe 'keystone::security_compliance' do
         is_expected.to contain_keystone_config('security_compliance/lockout_failure_attempts').with_value(3)
         is_expected.to contain_keystone_config('security_compliance/minimum_password_age').with_value(4)
         is_expected.to contain_keystone_config('security_compliance/password_expires_days').with_value(5)
-        is_expected.to contain_keystone_config('security_compliance/password_regex').with_value('SomeRegex')
-        is_expected.to contain_keystone_config('security_compliance/password_regex_description').with_value('this is some regex')
-        is_expected.to contain_keystone_config('security_compliance/unique_last_password_count').with_value(6)
+        is_expected.to contain_keystone_config('security_compliance/password_regex').with_value('^(?=.*\d)(?=.*[a-zA-Z]).{7,}$')
+        is_expected.to contain_keystone_config('security_compliance/password_regex_description').with_value('password must be at least 7 characters long and contain 1 digit')
+        is_expected.to contain_keystone_config('security_compliance/unique_last_password_count').with_value(2)
       end
     end
   end
-- 
2.34.1

